protocol-units/mempool/move-rocks/src/lib.rs

```diff
@@ -245,39 +273,71 @@ impl MempoolTransactionOperations for RocksdbMempool {
 		let db = self.db.clone();
 		tokio::task::spawn_blocking(move || {
 			let cf_handle = db
-				.cf_handle("mempool_transactions")
+				.cf_handle(cf::MEMPOOL_TRANSACTIONS)
 				.ok_or_else(|| Error::msg("CF handle not found"))?;
-			let mut iter = db.iterator_cf(&cf_handle, rocksdb::IteratorMode::Start);
-
+			let lookups_cf_handle = db
+				.cf_handle(cf::TRANSACTION_LOOKUPS)
+				.ok_or_else(|| Error::msg("CF handle not found"))?;
+			let mut iter = db.iterator_cf(&cf_handle, IteratorMode::Start);
+			let mut batch = WriteBatch::default();
 			let mut mempool_transactions = Vec::with_capacity(n as usize);
 			while let Some(res) = iter.next() {
 				let (key, value) = res?;
 				let transaction: MempoolTransaction = bcs::from_bytes(&value)?;
-				db.delete_cf(&cf_handle, &key)?;
 
-				// Optionally, remove from the lookup table as well
-				let lookups_cf_handle = db
-					.cf_handle("transaction_lookups")
-					.ok_or_else(|| Error::msg("CF handle not found"))?;
-				db.delete_cf(&lookups_cf_handle, transaction.transaction.id().to_vec())?;
+				batch.delete_cf(&cf_handle, &key);
+				batch.delete_cf(&lookups_cf_handle, transaction.transaction.id().to_vec());
 
 				mempool_transactions.push(transaction);
-				if mempool_transactions.len() > n - 1 {
+				if mempool_transactions.len() >= n {
 					break;
 				}
 			}
+
+			db.write(batch)?;
+
 			Ok(mempool_transactions)
 		})
 		.await?
 	}
+
+	async fn gc_mempool_transactions(&self, timestamp_threshold: u64) -> Result<(), anyhow::Error> {
+		let db = self.db.clone();
+		tokio::task::spawn_blocking(move || {
+			let cf_handle = db
+				.cf_handle(cf::MEMPOOL_TRANSACTIONS)
+				.ok_or_else(|| Error::msg("CF handle not found"))?;
+			let lookups_cf_handle = db
+				.cf_handle(cf::TRANSACTION_LOOKUPS)
+				.ok_or_else(|| Error::msg("CF handle not found"))?;
+			let mut read_options = ReadOptions::default();
+			read_options
+				.set_iterate_upper_bound(construct_timestamp_threshold_key(timestamp_threshold));
+			let mut iter = db.iterator_cf_opt(&cf_handle, read_options, IteratorMode::Start);
+			let mut batch = WriteBatch::default();
```

Comment:
So, I think ideally, we would be using [`DeleteRange`](https://rocksdb.org/blog/2018/11/21/delete-range.html) here.

Alternatively, we could build manual index by writing into GC designated column families. But, I don't think this is any better than the above.

Reply:
I thought of using that operation, but we also need to remove the "transaction_lookups" entries, and I don't see how we can do it without enumerating the removed transactions first.